### PR TITLE
feat(mailer): Resend handler

### PR DESCRIPTION
**Pull request by Josh-Walker-GM**
**Description**
This is intended to be a mail handler which specifically uses the [resend node sdk](https://github.com/resendlabs/resend-node/tree/main). 

**Changes**
1. Adds a new resend mail handler which uses the resend node sdk to send mail.
2. Removes an unneeded `uuid` dependency from the in-memory mail handler. I noticed this when copying over the handler as a template for this new one. 

**Notes**
1. I had to add a transformation of the attachment contents when they were passed as a string. If I did not transform them into a buffer then the attachment content would be malformed during my testing.
2. I have tested this manually but for the moment there is no specific testing suite/strategy in place to test these mail handlers/renderers.

### CODE REVIEW

File: packages/mailer/handlers/resend/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+  async send(
+    content: MailRenderedContent,
+    sendOptions: MailSendOptionsComplete,
+    handlerOptions?: ResendMailHandlerOptions
+  ): Promise<MailResult> {
+    // I was not having success at passing attachment contents as strings directly
+    // to the Resend client, so I'm going to transform them to Buffers if they are
+    // strings.
+    const transformedAttachments = []
```
Comment (dthyresson) [inline packages/mailer/handlers/resend/src/index.ts:31]: Do you think other mail handlers will have a similar need? If so, break out into a util function? Can do that refactor later perhaps, unless easier to test individual function to convert.

**Review by dthyresson — COMMENTED** [review]
See comments. Otherwise LGTM!

File: packages/mailer/handlers/resend/README.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# Mailer - Handler - Resend
+
+**WIP**: This package is still a work in progress.
```
[the remark this answers is not in the document]
Comment (Josh-Walker-GM) [reply packages/mailer/handlers/resend/README.md:3]: Okay great idea I'll add one here. It'll likely evolve again if we try and standardise the handler/renderer package readmes with some sort of template.

File: packages/mailer/handlers/resend/src/index.ts (hunk 1 above)
Comment (Josh-Walker-GM) [reply packages/mailer/handlers/resend/src/index.ts:31]: Not sure. In theory resend allows the string type but passing them in directly was confusing when it was malformed in my basic tests. 

Happy to either extract it out or even change the interface (not user facing but handler facing) to adapt to what we find more convenient as we add more handlers.

Comment (dthyresson) [reply packages/mailer/handlers/resend/src/index.ts:31]: Ah, ok. Let's refactor/revisit when implementing other handlers like Postmark, etc etc

File: packages/mailer/handlers/resend/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (Josh-Walker-GM) [reply packages/mailer/handlers/resend/README.md]: @dthyresson Let me know if the updated readme is okay. If so can you approve and then I'll merge.

[the remark this answers is not in the document]
Comment (dthyresson) [reply packages/mailer/handlers/resend/README.md]: Thanks that readme helps!

**Comment by dthyresson** [comment]
@Josh-Walker-GM approved and good to merge

**Pull request merged**